### PR TITLE
Move certain settings to the `Global` scope

### PR DESCRIPTION
**Pull request by Duhemm**
It was incorrectly unscoped since #714

### CODE REVIEW

**Comment by densh** [comment]
We also have to document scoping changes on the sbt page of the docs. This is going to be quite surprising for the end users otherwise.

**Comment by Duhemm** [comment]
I've been thinking a bit more about it and I think that instead I should improve scoping of configuration keys. For instance, I would expect

> set nativeMode := "debug"

To have an effect in the `compile` and `test` configurations. I would also like to be able to use the `release` mode in `compile` and `debug` in test.

What do you think of:
```
[info] Set current project to scala-native (in build file:/Users/martin/Documents/Projects.nosync/Duhemm/scala-native/)
> set nativeMode in sandbox := "release"
[info] Defining sandbox/*:nativeMode
[info] The new value will be used by sandbox/compile:nativeConfig, sandbox/test:nativeConfig
[info] Reapplying settings...
[info] Set current project to scala-native (in build file:/Users/martin/Documents/Projects.nosync/Duhemm/scala-native/)
> set nativeMode in Test in sandbox := "debug"
[info] Defining sandbox/test:nativeMode
[info] The new value will be used by sandbox/test:nativeConfig
```

**Comment by densh** [comment]
@Duhemm You suggestion sounds very reasonable to me. 👍

**Comment by Duhemm** [comment]
@densh I've implemented my suggestion. The settings in `globalSettings` now behave much like `scalacOptions`.

For instance, `set nativeMode in benchmarks := "release"` will have an effect when compiling and testing:

```
> set nativeMode in Test in benchmarks := "release"
[info] Defining benchmarks/test:nativeMode
[info] The new value will be used by benchmarks/test:nativeConfig
> set nativeMode in benchmarks := "release"
[info] Defining benchmarks/*:nativeMode
[info] The new value will be used by benchmarks/compile:nativeConfig
> set nativeMode in Compile in benchmarks := "release"
[info] Defining benchmarks/compile:nativeMode
[info] The new value will be used by benchmarks/compile:nativeConfig
> set nativeMode in benchmarks := "debug"
[info] Defining benchmarks/*:nativeMode
[info] The new value will be used by no settings or tasks.
> reload
> set nativeMode in benchmarks := "release"
[info] Defining benchmarks/*:nativeMode
[info] The new value will be used by benchmarks/compile:nativeConfig, benchmarks/test:nativeConfig
```

What do you think?

**Comment by densh** [comment]
@Duhemm What happens if you do not mention the scope at all? Does it still work? I think that it's way too easy to forget the scope and it should "just work" without it. I've made this mistake quite a few times during the development of 0.3 and was completely confused how come the settings are not being applied. We should aim for least confusion, and make sure scopes are optional feature for advanced users only.

**Comment by Duhemm** [comment]
@densh Yes it will work. Look at the last example after `reload` in the snippet I posted above.

For instance, to find the value to use for `nativeMode` in the `Test` configuration, sbt will use the first defined value of:

```
> inspect benchmarks/test:nativeMode
(...)
[info] Delegates:
[info]  benchmarks/test:nativeMode
[info]  benchmarks/runtime:nativeMode
[info]  benchmarks/compile:nativeMode
[info]  benchmarks/*:nativeMode
(...)
```

So the one defined in the `Global` (`*`) configuration will be used if no overriding value is found in the `Test`, `Runtime` or `Compile` configurations. That's exactly the behaviour of `scalacOptions` in sbt.

File: sbt-scala-native/src/main/scala/scala/scalanative/sbtplugin/ScalaNativePluginInternal.scala [hunk 1]
```diff
@@ -154,6 +155,14 @@ object ScalaNativePluginInternal {
       }
       libs
     },
+    nativeMode := "debug",
+    nativeLinkerReporter := tools.LinkerReporter.empty,
+    nativeOptimizerReporter := tools.OptimizerReporter.empty,
+    nativeLogger := streams.value.log,
+    nativeGC := "boehm"
+  )
```
Comment (densh) [inline sbt-scala-native/src/main/scala/scala/scalanative/sbtplugin/ScalaNativePluginInternal.scala]: It seems like all settings/keys that we document should be here. Why are `nativeCompileOptions`, `nativeLinkingOptions`, `nativeClang`, `nativeClangPP` not there?

Comment (Duhemm) [inline sbt-scala-native/src/main/scala/scala/scalanative/sbtplugin/ScalaNativePluginInternal.scala]: They are, I think you've been tricked by the folded diff. All the keys in global scope are:
`nativeWarnOldJVM`, `nativeClang`, `nativeClangPP`, `nativeCompileOptions`, `nativeLinkingOptions`, `nativeMode`, `nativeLinkerReporter`, `nativeOptimizerReporter`, `nativeLogger`, and `nativeGC`.

Comment (densh) [inline sbt-scala-native/src/main/scala/scala/scalanative/sbtplugin/ScalaNativePluginInternal.scala]: Indeed, folded diff is very misleading here.

**Comment by densh** [comment]
LGTM!

**Pull request merged**